lib/chef/provider/user/solaris.rb

```diff
@@ -46,7 +46,11 @@ def manage_user
         end
 
         def check_lock
-          shadow_line = shell_out!("getent", "shadow", new_resource.username).stdout.strip rescue nil
+          shadow_line = if node["platform"] == "solaris2"
+                          shell_out!("grep", new_resource.username, @password_file)
```

Comment:
What if you have a user named `greg` and a user named `gregory`?  Perhaps we should ancor the search a bit: `"^#{new_resource.username}:"` to ensure that we only pick up exact matches.  Alternatively, we could read in the whole file and parse it directly.